component/src/ibc/ibc_token.rs

```diff
@@ -0,0 +1,58 @@
+use ibc::core::ics24_host::identifier::{ChannelId, PortId};
+use penumbra_crypto::asset;
+
+/// IBC token respresents a token that was created through IBC.
+pub struct IBCToken {
+    transfer_path: String,
```

Comment:
Along the lines of [parse, don't validate](https://lexi-lambda.github.io/blog/2019/11/05/parse-don-t-validate/), maybe instead of storing the encoded `transfer_path`, we should store
```
channel_id: ChannelId,
port_id: PortId,
origin_denom: String,
```

so that we can access the channel and port IDs without re-parsing?